src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs

```diff
@@ -143,14 +190,181 @@ public class DependencyGraph
                 rootDependencies,
                 rootRepoFolder,
                 rootRepoCommit,
-                includeToolset,
+                options,
                 false,
                 logger,
                 reposFolder,
                 remotesMap,
                 testPath);
         }
 
+        private static void ValidateBuildOptions(
+            IRemoteFactory remoteFactory,
+            IEnumerable<DependencyDetail> rootDependencies,
+            string repoUri,
+            string commit,
+            DependencyGraphBuildOptions options,
+            bool remote,
+            ILogger logger,
+            string reposFolder,
+            IEnumerable<string> remotesMap,
+            string testPath)
+        {
+            // Fail fast if darcSettings is null in a remote scenario
+            if (remote && remoteFactory == null)
+            {
+                throw new DarcException("Remote graph build requires a remote factory.");
+            }
+
+            if (rootDependencies != null && !rootDependencies.Any())
+            {
+                throw new DarcException("Root dependencies were not supplied.");
+            }
+
+            if (!remote)
+            {
+                if (options.LookupBuilds)
+                {
+                    throw new DarcException("Build lookup only available in remote build mode.");
+                }
+                if (options.NodeDiff != NodeDiff.None)
+                {
+                    throw new DarcException($"Node diff type '{options.NodeDiff}' only available in remote build mode.");
+                }
+            }
+            else
+            {
+                if (options.NodeDiff != NodeDiff.None && !options.LookupBuilds)
+                {
+                    throw new DarcException("Node diff requires build lookup.");
+                }
+            }
+        }
+
+        private static async Task DoLatestInChannelGraphNodeDiffAsync(
```

Comment:
>DoLatestInChannelGraphNodeDiffAsync [](start = 34, length = 35)

Do we really need all these bells and whistles? Are people asking for them?  Can we just choose one behavior for now and go with that to avoid all this complexity we might never need?

Reply:
We do. These have all been added because of information I've needed to gather as I've shutdown the previews.